source/extensions/request_id_utils/uuid/uuid_impl.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include "envoy/request_id_utils/request_id_utils.h"
+
+#include "common/runtime/runtime_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace RequestIDUtils {
+class UUIDUtils : public Envoy::RequestIDUtils::Utilities {
+public:
+  UUIDUtils(Envoy::Runtime::RandomGenerator& random) : random(random) {}
+
+  void setRequestID(Http::HeaderMap& request_headers);
+  void ensureRequestID(Http::HeaderMap& request_headers);
+  bool modRequestIDBy(const Http::HeaderMap& request_headers, uint64_t& out, uint64_t mod);
+  Envoy::RequestIDUtils::TraceStatus getTraceStatus(const Http::HeaderMap& request_headers);
+  void setTraceStatus(Http::HeaderMap& request_headers,
+                      const Envoy::RequestIDUtils::TraceStatus status);
+
+private:
+  // Reference to the random generator used to generate new request IDs
+  Envoy::Runtime::RandomGenerator& random;
+
+  // Byte on this position has predefined value of 4 for UUID4.
+  static const int TRACE_BYTE_POSITION = 14;
```

Comment:
nit: maybe move this and other constants below to anonymous namespace in `.cc` file?